test/conversion_test.go

```diff
@@ -0,0 +1,978 @@
+    - name: kind
+      type: string
+      value: Task
+  timeouts:
+    pipeline: 60s
+  serviceAccountName: default
+`
+
+	v1PipelineRunYamlAlpha = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  pipelineRef:
+    resolver: "bundles"
+    kind: Pipeline
+    params:
+    - name: bundle
+      type: string
+      value: "test-bundle"
+    - name: name
+      type: string
+      value: pr
+    - name: kind
+      type: string
+      value: Task
+  timeouts:
+    pipeline: 60s
+  taskRunTemplate: {
+    serviceAccountName: default
+  }
+`
+)
+
+// TestTaskCRDConversion first creates a v1beta1 Task CRD using v1beta1Clients and
+// requests it by v1Clients to compare with v1 if the conversion has been
+// correctly executed by the webhook. And then it creates the v1 Task CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestTaskCRDConversion(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	t.Parallel()
+
+	c, namespace := setup(ctx, t)
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
+
+	v1beta1TaskName := helpers.ObjectNameForTest(t)
+	v1beta1Task := parse.MustParseV1beta1Task(t, fmt.Sprintf(v1beta1TaskYaml, v1beta1TaskName, namespace))
+	v1TaskExpected := parse.MustParseV1Task(t, fmt.Sprintf(v1TaskYaml, v1beta1TaskName, namespace))
+
+	if _, err := c.V1beta1TaskClient.Create(ctx, v1beta1Task, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create v1beta1 Task: %s", err)
+	}
+
+	v1TaskGot, err := c.V1TaskClient.Get(ctx, v1beta1TaskName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected v1 Task for %s: %s", v1beta1TaskName, err)
+	}
+
+	if d := cmp.Diff(v1TaskExpected, v1TaskGot, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+
+	v1TaskName := helpers.ObjectNameForTest(t)
+	v1Task := parse.MustParseV1Task(t, fmt.Sprintf(v1TaskYaml, v1TaskName, namespace))
+	v1beta1TaskExpected := parse.MustParseV1beta1Task(t, fmt.Sprintf(v1beta1TaskYaml, v1TaskName, namespace))
+
+	if _, err := c.V1TaskClient.Create(ctx, v1Task, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create v1beta1 Task: %s", err)
+	}
+
+	v1beta1TaskGot, err := c.V1beta1TaskClient.Get(ctx, v1TaskName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected v1beta1 Task for %s: %s", v1TaskName, err)
+	}
+
+	if d := cmp.Diff(v1beta1TaskExpected, v1beta1TaskGot, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+}
+
+// TestTaskRunCRDConversion first creates a v1beta1 TaskRun CRD using v1beta1Clients
+// and requests it by v1Clients to compare with v1 if the conversion has been correctly
+// executed by the webhook. And then it creates the v1 TaskRun CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestTaskRunCRDConversion(t *testing.T) {
```

Comment:
Sorry for the very late check, leaving this comment to track: we need to update `TestTaskRunCRDConversion`, `TestTaskRunCRDConversionAlpha`, `TestPipelineRunCRDConversion` and `TestPipelineRunCRDConversionAlpha`:
1. Add status to all **expected** YAMLs of runtime objects -- PipelineRun and TaskRun.
2. Create `v1beta1` Object, wait until it finishes (can use [updated](https://github.com/tektoncd/pipeline/issues/5793) version of [WaitForPipelineRunState](https://github.com/tektoncd/pipeline/blob/b3e4f35712fca0235e3ef80cc3f03b2600f40cb1/test/wait.go#L159) and [WaitForTaskRunState](https://github.com/tektoncd/pipeline/blob/b3e4f35712fca0235e3ef80cc3f03b2600f40cb1/test/wait.go#L85)), compare **finished** TaskRuns and PipelineRuns (of both `v1` and `v1beta1` versions) with **expected** objects.
3. [optional but highly recommended] create `v1` Objects, wait until it finishes and compare.

Reply:
Thanks @XinruZhang , this is very helpful.
Added to https://github.com/tektoncd/pipeline/pull/5579#issuecomment-1326617948.